### DISCUSSION TITLE
How can we remove and or at least change the `key`?

### DISCUSSION THREAD

**Discussion by kireerik** [emotion-js/emotion#3044] [General]
related documentation:

- https://emotion.sh/docs/install#vanilla

- https://emotion.sh/docs/ssr#extractcritical

- https://emotion.sh/docs/@emotion/cache#key

&nbsp;
&nbsp;
This is my current workaround:
- https://github.com/withastro/astro/issues/4432#issuecomment-1567871654

**Comment by Andarist** [comment]
I think that you are looking for `@emotion/css/create-instance`. This allows you to customize the `key`.

**Reply by kireerik** [reply]
Yes, thank you!

**Reply by kireerik** [reply]
First it seemed to work, but I think only because the local development server I am using was running and I tried `css` as the key which is the same as the default. So now I noticed that it is not working.

`extractCritical` responds with empty `ids` and `css` (and only `html` is present on the response).

This is how I am trying it:
```Node.js
import createEmotion from '@emotion/css/create-instance'
const {css} = createEmotion({key: 'key'})

, tag = css`
	border: 1px solid;
`

import {extractCritical} from '@emotion/server'

console.log(extractCritical(tag))
```

**Reply by kireerik** [reply]
related issue:
- #2118

**Comment by kireerik** [comment]
For `extractCritical` do I need to provide HTML (string)?

&nbsp;
Just providing the class names separated by a space character seems to work fine (and I don't need the `html` result that `extractCritical` gives).

&nbsp;
> See the same example:
- https://github.com/withastro/astro/issues/4432#issuecomment-1567871654

**Reply by Andarist** [reply]
`extractCritical` expects the rendered HTML as its argument. It might work with other input but that's not 100% guaranteed.

**Comment by kireerik** [comment]
closed by #2118
